### DISCUSSION TITLE
UV Uniformity Test

### DISCUSSION THREAD

**Discussion by stickninja** [sn4k3/UVtools#916] [Ideas]
I've been thinking about tests someone can do before they even pour their first bottle of resin to determine if their specific printer sample is worth keeping. I know we all want better hardware, but the reality is, the manufacturers are selling a ton of printers to people who don't know how to test or what to look for. This has gotten worse with the new crop of auto leveling printers. Almost every day I'm helping someone new with problems with a Saturn 4 (most common auto-leveling printer I see). So my 2 basic checks have been: 1) check flatness of the build plate with a ruler or straight edge (a couple of users have caught plates with low spots doing this and requested replacements). 2) run a dry print run level test using a sheet of paper cut into 4 pieces. 1 under each corner. Normally not necessary with a manually leveled plate, although can be helpful if new people haven't leveled correctly (like adding a z offset that's unnecessary). For auto leveling printers this almost always catches a leveling problem with the system. On Saturn 4s/Mars 5s, this either means a replacement or adjusting the screws on top of the plate (Elegoo's current fix as they were replacign too many plates). Another user had an M5S that Anycubic told him to use a bottom exposure of 80s, and nothing still would adhere. Paper test showed the plate wasn't moving down far enough.

So, this leads me to my next idea. Some advanced users are using Chitu's UV light meter to test the uniformity of their LCD/light source. Jan Mrazek built a device he called DrLCD to do this with more data points. Another user I talked to simply took a picture with his digital camera, isolated the Blue channel and used that to create a mask in UVTools (then found out a Chitu printer couldn't handle a mask of that resolution). I was wondering if it would be possible for a user to take a picture with their smartphone, and then open the picture in UVTools and have a tool to isolate the B values and run a uniformity comparison, and maybe even write the values to a CSV. This doesn't have to be super precise, as we are looking just to see how much variation is across the screen. Using a Chitu meter with 9 or so data points, users are seeing as low as 75% uniformity, even though some manufacturers claim 95%. Phrozen even told one user that this was within spec. If we could achieve similar results using a phone camera, which most people have, it would be a valuable tool. We could even still use it to create an exposure mask in UVTools.

**Comment by sn4k3** [comment]
One can try with current mask tool, it allow to import a png. Use your smartphone to capture the image, crop it and resize to match your LCD size. However you will find it will not be precise and in process of resize you will lose some information.

My advice:

1. Place a paper sheet on LCD
2. Turn off lights and make this in pure dark
3. Turn on LCD and take the picture without flash in the most stable and straight as you can. (Do not use digital zoom, try to catch the most area in the frame).
4. Do the crop and resize
5. Mask with UVtools and do some tests.

After some tries you may find your optimal workflow and mask.
You will also find that probably you need to use manual camera settings rather than auto.
Make sure to focus the middle point.
